src/components/badge/badge.spec.tsx

```diff
@@ -54,4 +61,79 @@ describe("Badge", () => {
       expect(wrapper.find(BADGE).exists()).toBe(false);
     });
   });
+
+  describe("Accessibility", () => {
+    const ariaLabelText = "Generic aria message";
+    const wrapper = mount(
+      renderComponent({ counter: 9, "aria-label": ariaLabelText })
+    );
+
+    it("should have the relevant aria-label when aria-label is specified", () => {
+      expect(wrapper.find(Badge).first().prop("aria-label")).toBe(
+        ariaLabelText
+      );
+    });
+  });
+
+  describe("Styles", () => {
+    it("should apply the correct style when onClick is present", () => {
+      const wrapper = mount(renderComponent({ counter: 9, onClick: () => {} }));
+      assertStyleMatch(
+        {
+          padding: `0`,
+          width: `22px`,
+          minHeight: `22px`,
+          borderRadius: `50%`,
+          overflow: `hidden`,
+          textAlign: `center`,
+          position: `absolute`,
+          top: `-11px`,
+          right: `-11px`,
+          marginRight: `0`,
+          background: `var(--colorsActionMajorYang100)`,
+        },
+        wrapper.find(StyledButton)
+      );
+    });
+
+    it("should apply the correct hover style when onClick is present", () => {
+      const wrapper = mount(renderComponent({ counter: 9, onClick: () => {} }));
+      assertStyleMatch(
+        {
+          background: `var(--colorsActionMajor500)`,
+          border: `none`,
+        },
+        wrapper.find(StyledButton),
+        {
+          modifier: "&:hover",
+        }
+      );
+    });
+
+    it("should apply the correct style when onClick is not specified", () => {
+      const wrapper = mount(renderComponent({ counter: 9 }));
+      assertStyleMatch(
+        {
+          cursor: `default`,
```

Comment:
As above, I'm not sure if it makes sense to test styles here

Reply:
As per Ed's comment, I'll leave a test for cursor behaviour as Chromatic won't capture that until we implement the Storybook Play functions.